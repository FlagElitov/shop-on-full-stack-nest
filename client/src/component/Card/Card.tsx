import React from "react";
import ButtonComponent from "./Button/Button";

interface CardProps {
  name: string;
  balance: number;
  id: number;
  toggleHandler: (id: number) => void;
}

const Card: React.FC<CardProps> = ({ name, balance, id, toggleHandler }) => {
  return (
    <div className="card">
      <div className="name"> {name}</div>
      <div className="text"> Balance</div>
      <div className="balance">{balance}$</div>

      <div className="button">
        {balance < 100 ? (
          <div className="warning">Нет баланса!</div>
        ) : (
          <ButtonComponent toggleHandler={toggleHandler} id={id} />
        )}
      </div>
    </div>
  );
};

export default Card;
