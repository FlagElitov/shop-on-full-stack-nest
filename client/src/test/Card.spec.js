import { shallow } from "enzyme";
import React from "react";
import Card from "../component/Card/Card";

const setUp = () => shallow(<Card name="Petro" balance="500" id="2" />);

describe("Unit Testing App component", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });
  it("App Snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
