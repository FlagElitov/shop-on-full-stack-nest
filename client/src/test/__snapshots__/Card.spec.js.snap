// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing App component App Snapshot 1`] = `
<div
  className="card"
>
  <div
    className="name"
  >
     
    Petro
  </div>
  <div
    className="text"
  >
     Balance
  </div>
  <div
    className="balance"
  >
    500
    $
  </div>
  <div
    className="button"
  >
    <ButtonComponent
      id="2"
    />
  </div>
</div>
`;
